### DISCUSSION TITLE
Filtering multiple relation

### DISCUSSION THREAD

**Discussion by zakkor** [pocketbase/pocketbase#2367] [Q&A]
Let's imagine we have a collection called `tasks`, which has a field `assignedUsers` that is a multiple relation to `users`

We have two `users`, with IDs `U1` and `U2`

We have three `tasks`:
- T1 has assignedUsers: `[U1]`
- T2 has assignedUsers: `[U1, U2]`
- T3 has assignedUsers: `[U1, U2]`

Now let's say we want to get all `tasks` that have assignedUsers: `[U1, U2]`

I expected to be able to chain multiple uses of `?=` together in order, but it doesn't seem to work in the web UI:
- `assignedUsers.id ?= "U1"` returns T1, T2, T3 (as expected)
- `assignedUsers.id ?= "U2"` returns T2, T3 (as expected)

But then chaining them together:
`assignedUsers.id ?= "U1" && assignedUsers.id ?= "U2"` returns nothing, where I expected it to return T2, T3

Is this a bug? Or is there another way to do this filter?

**Comment by ganigeorgiev** [comment]
This is expected although I agree that it may not be very clear.

`assignedUsers.id` represents only a single value and it cannot be two different things at the same time. You can think of it as a column value in a joined SQL row, something like (_simplified pseudo-sql_):

```sql
SELECT *
FROM tasks
LEFT JOIN assignedUsers
WHERE assignedUsers.id = "U1"
```


Array `IN`-like expressions at the moment are not supported and are in my todo, but for now are left aside since there are some complications with the multimatch/matchall handling.

If you want to check whether at least one of the ids exists, you can use the `||` operator.

If you want to check whether **both** ids exists at the same time, you can search within the serialized json array value with the `~` operator, something like:
```js
assignedUsers ~ '"UI1"' && assignedUsers ~ '"UI2"'
```
